### DISCUSSION TITLE
Releasing LXQt 0.17.0

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#1968] [General]
Dear LXQt members and contributors,

As we'd agreed on regular releases, LXQt 0.17.0 should be published some time in this month. Some PRs need to be reviewed and merged, translations need to be updated, and a small change needs to be reversed. Then, we/I need a week or so to test all components by using them.

I want to start releasing LXQt around April 15-17th. The process will be straightforward: ChangeLogs will be updated before the release, I'll put a draft of the announcement here for you to correct it if needed, and merges will be stopped during the release (which should be done in a day or two).

If, for whatever reason, that time isn't good for you, please tell me!

**Comment by stefonarch** [comment]
I'm ok with everything, it's you who does the work, thanks!

**Reply by tsujan** [reply]
BTW, it would be great if two of us make sure that their whole LXQt installations are based on the latest git sources. Two pairs of eyes are always better.

@stefonarch, do you have time for that? I need to upgrade only 2 or 3 components after the recent code cleanups.

Also, we need to reverse https://github.com/lxqt/lxqt-session/commit/b4ae46c2e5fdb19f45bb4177f8d0c37ef7b237d1; it breaks UX.

**Comment by stefonarch** [comment]
I just recompiled  this morning, let me know when last stuff is done.

**Reply by tsujan** [reply]
@stefonarch 
In case you haven't already done it, whenever you find the time, please compile and install the latetst git sources of following components in this order and then log out and log in back:
```
lxqt-build-tools
libqtxdg
liblxqt
libfm-qt
pcmanfm-qt
lxqt-qtplugin
lximage-qt
lxqt-archiver
```
I bumped their versions yesterday and they depend on each other.

I did it here minutes ago and everything seems fine but better safe than sorry. Tell me if you see a problem.

**Reply by stefonarch** [reply]
Compiling and re-login: no issue found, despite archiver throws a lot of warnings during compiling but this was so before too I think.

**Reply by tsujan** [reply]
> Compiling and re-login: no issue found

Thanks!

> despite archiver throws a lot of warnings during compiling but this was so before too I think.

Yes! They're harmless. Silencing them needs time. I might do it after LXQt 0.17.

**Comment by luis-pereira** [comment]
It's a perfect time.

**Reply by tsujan** [reply]
@luis-pereira 
Now that you're here, could you please take a look at https://github.com/lxqt/lxqt-session/pull/373 before the release day?

EDIT: Never mind! It was approved and merged.

**Reply by luis-pereira** [reply]
@tsujan Only saw this now.

**Comment by luis-pereira** [comment]
@tsujan What needs to be done for the release ?

**Reply by tsujan** [reply]
> What needs to be done for the release ?

Apart from updating translations (by @stefonarch) and updating changelogs and bumping versions (by me), nothing special.

You have 2 PRs that will be merged.

**Reply by tsujan** [reply]
Oh, I forgot something about your question:

After the release is made, https://lxqt.github.io/ needs to be updated based on the release announcement (whose draft I'll add here and whose completed form I'll add to https://github.com/lxqt/lxqt/releases) and, probably, changelogs.

I think you (@luis-pereira) did it with the previous release and I hope you'll have time for it in the case of 0.17; if not, please tell me.

**Reply by stefonarch** [reply]
Btw, any news about accessing our stale twitter account  @jubalh ?

**Reply by jubalh** [reply]
@stefonarch I tried to contact Simon several times. Unfortunately didn't hear back. If I remember correctly Alf also had access to it. Quite sad we can't use it to propagate news :-/ I'll be ready to package for openSUSE once release. And will share that on twitter at least.

**Reply by tsujan** [reply]
@stefonarch 
Good that you asked that question from @jubalh. I remembered that I wanted to ask you what should be added to the "General" section of the announcement about the work that you, @yan12125 and other members/contributors did about everything other than LXQt components (the site, donations, etc.). Of course, you could edit that section when the draft is added here but could you give me a general description?

**Reply by jubalh** [reply]
I just wrote to Simon one more time. Let's see whether it helps.. @tsimonq2 ping

**Reply by stefonarch** [reply]
Those come to my mind:
 * Donations IMHO is not ready for announcement until we don't have more members that just one in the "team" of opencollective, some more clear statements how to use them ecc.
 * many translations completed, 100+ translators in weblate
 * The "matrix-telegram-IRC" bridged messenger opportunities
 * "Discussions" in GH

**Reply by yan12125** [reply]
Regarding donations, I was hoping that it would be ready for an announcement at the time of 0.17 release. More administrators for managing donations can be done in a short time if anyone is willing to take the role. I will still actively handle affairs around donations when there are more administrators. Improving  statements for donations may take longer and is beyond my ability as it is more like marketing rather than explaining technical stuffs.

**Reply by stefonarch** [reply]
> More administrators for managing donations can be done in a short time if anyone is willing to take the role.

I think only way is invitation ;)
@jubalh @luis-pereira @palinek @wxl 

I will join too if there are more, but atm I'm the only one to file reimbursements...

**Reply by telmobss1** [reply]
Hi guys, starting to release the new update, can send me a draft of the biggest news? 

I'm still waiting to use the official Twitter of the lxqt project

**Reply by telmobss1** [reply]
> Em relação às doações, esperava que estivesse pronto para um anúncio na hora do lançamento do 0.17. Mais administradores para gerenciar doações podem ser feitos em um curto espaço de tempo se alguém estiver disposto a assumir a função. Ainda cuidarei ativamente de questões relacionadas a doações quando houver mais administradores. Melhorar as declarações de doações pode levar mais tempo e está além da minha capacidade, pois é mais como marketing do que explicar coisas técnicas.

I wouldn't mind taking care of it, but I need documentation first, and I think there's still a lot of doubt about it.

**Reply by luis-pereira** [reply]
> I think you (@luis-pereira) did it with the previous release and I hope you'll have time for it in the case of 0.17; if not, please tell me.

I will do it.

**Reply by luis-pereira** [reply]
> You have 2 PRs that will be merged.

Just merged them (`lxqt-policykit`).

**Reply by tsujan** [reply]
> Hi guys, starting to release the new update, can send me a draft of the biggest news?

@telmo1241
The announcement will appear at https://github.com/lxqt/lxqt/releases and, before that, its draft will be added to the current page for other members to edit it. I'm making changelogs and preparing the announcement based on them.

Since some components might change before the release (e.g., `lxqt-session` might include a big change), the draft will need a little more time to be finished.

**Reply by telmobss1** [reply]
> > Hi guys, starting to release the new update, can send me a draft of the biggest news?
> 
> @telmo1241
> The announcement will appear at https://github.com/lxqt/lxqt/releases and, before that, its draft will be added to the current page for other members to edit it. I'm making changelogs and preparing the announcement based on them.
> 
> Since some components might change before the release (e.g., `lxqt-session` might include a big change), the draft will need a little more time to be finished.

Okay, what about lxqt's Twitter?

**Reply by tsujan** [reply]
> Okay, what about lxqt's Twitter?

→ https://github.com/lxqt/lxqt/discussions/1968#discussioncomment-580883

**Reply by yan12125** [reply]
> I wouldn't mind taking care of it, but I need documentation first, and I think there's still a lot of doubt about it.

@telmo1241 For doubts, I think I answered at https://github.com/lxqt/lxqt/issues/1846. Could you also point out what kind of documentation is missing at that issue?

**Reply by tsujan** [reply]
This is a draft of the main sections of the release note. Feel free to edit it if something is missing or wrong.

# Highlights

## General

 * LXQt terminates all child processes on exiting the session. This allows non-LXQt apps to save their last settings at the session end and also prevents rare exit crashes in some of them.
 * Creation time is fully supported by the file manager.
 * SVG icon sets are handled more efficiently and independently.
 * LXQt Power Manager has separate idle watchers for AC and battery and also an option to disable them when the active window is fullscreen.
 * LXQt Panel has an option for behaving like a dock, i.e., auto-hiding only on overlapping a window.
 * Translations are completely done through Weblate and are constantly updated for all components.
 * GitHub's [Discussions section for LXQt](https://github.com/lxqt/lxqt/discussions) is up and running. Asking questions, making suggestions and discussing ideas can be done most efficiently by using "Discussions". Also, the "matrix-telegram-IRC" bridged messenger has become possible.

## LibFM-Qt / PCManFM-Qt

 * The creation time is fully supported.
 * An item is added to Tools menus to activate Admin mode provided by GVFS — for transferring files beyond user permissions but without a root instance. *(NOTE: Admin is broken in GVFS 1.48.0; it will be enabled automatically whenever its bug is fixed by GVFS devs.)*
 * Launchers can be created from Tools menu (and not only on Desktop).
 * Better opening of a mixed selection of files that have different mime types.
 * Natural keyboard navigation on Desktop.
 * File size limits for thumbnail creation are activated.
 * Blurred thumbnails with scale factors > 1 are fixed.
 * The localization of LXQt file dialog is enabled.
 * Various enhancements and fixes.

## LXQt Panel

 * An option is added to auto-hide a panel only when it overlaps a window.
 * A problem is fixed in changing of world clock's time zone with mouse wheel.
 * The grouped task button popup is fixed in a special case.

## QTerminal / QTermWidget

 * "Action after paste" is made clearer and its default is changed to "Scrolling to bottom".
 * Added 5 modes for background images. NOTE: Since background images are drawn differently, manual reconfiguration may be needed; see docs/configuration.md for more details.
 * Added an option to disable bracketed paste mode.

## LXQt Power Management

 * Idle watchers of AC and battery are separated from each other.
 * An option is added to disable idle watcher when the active window is fullscreen.

## LXImage Qt

 * Fitting images on navigating is made optional.
 * Thumbnail options are added to Preferences dialog.

## LXQt Archiver

 * Disk images can be opened or extracted.
 * Window settings are remembered.
 * The side-pane is made horizontally scrollable.

## libQtXdg

 * SVG icon sets are handled more completely and independently by using Qt's SVG renderer.
 * XTerm is added as a runtime dependency.

## LXQt Desktop Notifications

The notification summary is always handled as plain-text.

## LXQt Session

 * All child processes are terminated at the session end.
 * Openbox configuration is removed for the sake of WM agnosticism.

## Qps

Fixed a crash on showiing Qps' window if Qps is started minimized to tray.

**Reply by stefonarch** [reply]
* Panel: "Put buttons of the same class next to each other" was there in 0.16 already?
* "Copy result" in runner is new
* Specify "creation time" only on supported filesystems ?

**Reply by tsujan** [reply]
> Panel: "Put buttons of the same class next to each other" was there in 0.16 already?

Yes, it was.

> "Copy result" in runner is new

It will be in Runner's release note. Here, we usually mention the most important changes.

> Specify "creation time" only on supported filesystems ?

It's a part of supporting creation time, although it had a separate commit.

**Reply by tsujan** [reply]
More specifically, I first make release notes for all components and then extract the most important news for the announcement. Release notes themselves are extracted from changlogs, which I update in "prerelease" branches.

**Reply by tsujan** [reply]
OK, all "pre-release" changes are made. I added a line to announcement about fixing Qps' crash. Releasing will be just a matter of copy-pasting, clicking and pushing buttons :)

**Reply by stefonarch** [reply]
Give me just a ping when to commit and merge the last translating contributions. Still much activity atm for dutch and estonian.

**Reply by tsujan** [reply]
> Give me just a ping when to commit...

Sure.

**Reply by tsujan** [reply]
@stefonarch 
Today, I'll start the actual release. Please merge translation PRs if there are any.

**Reply by stefonarch** [reply]
All done, go and click :)

**Reply by tsujan** [reply]
Done.

As always, if you found a problem, please tell me to fix it.

**Reply by tsujan** [reply]
@luis-pereira
 https://lxqt.github.io/ can be updated — whenever you find the time.

**Comment by rubyFeedback** [comment]
Could someone add a link or something to "what's new"?

I had a look at:

https://lxqt-project.org/ but there is no news in this regard. I only saw this all on github.

I understand that github is more convenient for the devs, but Average Joe (and I am
one of them) might like to see a more "centralized" way to read what's new, a bit like
how xfce does it too (although they also rarely update the main homepage... 
but another example is the gtk blog; I don't see a link to the blog on the main page
of lxqt, and it does not mention the new 0.17.x release either...)

**Reply by luis-pereira** [reply]
@tsujan @rubyFeedback I will update the site.

**Reply by tsujan** [reply]
@rubyFeedback
The release was made only < 2 hours ago. Updating the site will take some time.

**Reply by tsujan** [reply]
I'll make point releases of `libqtxdg`, `libfm-qt`,` lxqt-config`, `lxqt-session` and `lxqt-panel` tomorrow. They're only for Qt < 5.15 and, especially, for Qt5.12; distros like Arch and OpenSUSE won't see a change.

**Reply by tsujan** [reply]
https://lxqt-project.org has been updated by @luis-pereira.

**Comment by darkshram** [comment]
Everything goes fine with LXQt 0.17.0 built with Qt 5.12.10 + patches.

![screen](https://user-images.githubusercontent.com/10327500/114944084-9dd96b80-9e0c-11eb-928d-9a11376ccb5c.jpg)

![screen2](https://user-images.githubusercontent.com/10327500/114944095-a2058900-9e0c-11eb-8945-6e9611e1c1f1.jpg)

All patches done today worked fine.

The only relevant messages in ~/.xsession-errors are:

```
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: cHRM chunk does not match sRGB
QToolButton(0x20ca2c0, name="closeButton")  does not have a property named  "icon-size"
QToolButton(0x1fffe10, name="closeButton")  does not have a property named  "icon-size"
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 3941, resource id: 69206172, major code: 18 (ChangeProperty), minor code: 0
libpng warning: iCCP: known incorrect sRGB profile
QToolButton(0x1feaf70, name="closeButton")  does not have a property named  "icon-size"
libpng warning: iCCP: known incorrect sRGB profile
QToolButton(0x1f17d20, name="closeButton")  does not have a property named  "icon-size"
QToolButton(0x1ff1580, name="closeButton")  does not have a property named  "icon-size"
QToolButton(0x1f19b20, name="closeButton")  does not have a property named  "icon-size"
QToolButton(0x20503a0, name="closeButton")  does not have a property named  "icon-size"
QToolButton(0x1f3d1c0, name="closeButton")  does not have a property named  "icon-size"
```
Theory:  there are some icons somewhere made with Photoshop with wrong iCCP chunk.

**Reply by tsujan** [reply]
"qt.qpa.xcb: QXcbConnection: XCB error..." is a debug message that may appear now and then with Qt + X11. It's harmless.

The libpng warning isn't specific to Qt and can be ignored.

The other warning about QToolButton can be silenced.

Thank you for your complete tests with Qt 5.12!

**Reply by darkshram** [reply]
Just as a footnote: I have read at [Qt forums](https://forum.qt.io/topic/58638/solved-libpng-warning-iccp-known-incorrect-srgb-profile-drive-me-nuts/4) a few years back about `libpng warning: iCCP`. Means there are some PNG images/icons somewhere within some Qt5[core|related] components which were created or edited with a buggy version of Photoshop. I remember testing doing `find /usr/lib64/qt5 -type f -name "*.png" |xargs -I{} sudo optipng "{}"` and that made all these warnings disappear (optipng fixes the defective iCCP chunk).

**Reply by tsujan** [reply]
You mean `/usr/share/icons/hicolor/32x32/apps/screengrab.png`? Here, the icon theme is used instead of it. Screengrab was imported into LXQt. If the warning is about that image and you can silence it, please make a PR at https://github.com/lxqt/screengrab.

**Reply by darkshram** [reply]
> You mean `/usr/share/icons/hicolor/32x32/apps/screengrab.png`? Here, the icon theme is used instead of it. Screengrab was imported into LXQt. If the warning is about that image and you can silence it, please make a PR at https://github.com/lxqt/screengrab.

No need. The warnings are generated because somewhere within the filesystem there are PNG images with a _Photoshop profile_ used by Qt5 components. Screengrab icon has no issues (icon is already optimized). The warning maybe silenced if every Photoshop-edited-PNG image used by Qt is fixed.

**Reply by tsujan** [reply]
Well, as a *user*, I don't like such messages to be created on my hard drive constantly. So, I've symlinked `~/.local/share/sddm/xorg-session.log` to `/dev/null` and created `~/.config/QtProject/qtlogging.ini` as:
```
[Rules]
*.warning=false
```
I've also set `MaxLevelStore`, `MaxLevelSyslog` and `MaxLevelKMsg` to `warning` in `/etc/systemd/journald.conf`.

As a dev, I temporarily remove the contents of `~/.config/QtProject/qtlogging.ini` and use a terminal whenever I want to see Qt's probable messages with specific apps — they can be important if existing and used in this way.

**Comment by jubalh** [comment]
@tsujan I think i spotted a mistake: https://github.com/lxqt/lxqt-config/releases/tag/0.17.0 the tarball is called lxqt-config-0.17.1.tar.xz instead of lxqt-config-0.17.0.tar.xz

**Reply by tsujan** [reply]
@jubalh 
Thanks a lot!

Minutes ago, I released 0.17.1 to fix compilation against Qt 5.12 (→ https://github.com/lxqt/lxqt-config/releases/tag/0.17.1). Do I need to fix the name of 0.17.0 package after that?

**Reply by jubalh** [reply]
Ah that's great. I think it's not important that. Though it might be that it confuses some people. But I assume most will check latest only. Thanks!

**Reply by tsujan** [reply]
> Though it might be that it confuses some people

You're right. I'll fix it.

EDIT: Done. It was a little tricky but I leaned something new from it :)

**Comment by jubalh** [comment]
I finished packaging LXQt 0.17.0 for openSUSE. It will go to QA and then hopefully be in Tumbleweed soon! Thanks for all the work!

**Reply by tsujan** [reply]
It only remains to release 0.17.1 for `lxqt-panel` and `lxqt-session` but they will make no difference for Tumbleweed at all.

**Reply by jubalh** [reply]
Yep I read it. I'll package them later anyways just so that we use the latest numbers.

**Reply by tsujan** [reply]
With lxqt-session-0.17.1, my job is finished. @stefonarch, translation PRs are free to come and be merged. Distro devs can make packages (already done by @jubalh for Tumbleweed).

Let's work toward the next release, which might be 1.0.0.

**Comment by darkshram** [comment]
Some of the AUTHORS files still have `https://lxqt.org` instead of `https://lxqt.github.io`.

**Reply by tsujan** [reply]
They should be changed to https://lxqt-project.org/. PRs would be appreciated.

**Reply by darkshram** [reply]
There is another issue. The `razor-qt.org` domain referenced in many AUTHORS files has a new owner and the website displays something that has nothing to do with defunct Razor-Qt or FLOSS.

**Reply by tsujan** [reply]
Yes, the site was removed a long time ago and now, it's about something else.

I guess we should remove all traces of `razor-qt.org` but (1) that requires the approval of LXQt members and (2) I don't know how we could give credit to the old razor group — or can we forget about that after several years?

**Reply by stefonarch** [reply]
Replace the link with https://en.wikipedia.org/wiki/Razor-qt ?
 I found 16 files.

**Reply by stefonarch** [reply]
As for lxqt.org there are 667 files..

**Reply by darkshram** [reply]
I think it would be better to replace `http://razor-qt.org` with `https://github.com/Razor-qt/razor-qt`. I think that won't require any approval at all from all developers.

**Reply by stefonarch** [reply]
O sure that's better, wasn't aware that there is still a repo (archived).

**Comment by nbvcxz** [comment]
Works really great, but you have to improve application menu and provide more themes (consistent with most popular GTK+ and QT) to gain more attention. Great job and thank you!
![image](https://user-images.githubusercontent.com/4711565/115062309-81abfc00-9eea-11eb-952d-27ca24a7d04b.png)

**Reply by jubalh** [reply]
Would be great if more volunteers could create themes. Are you good at it? Do you know people who are good at design? Let's convince them to join!

**Reply by nbvcxz** [reply]
So far I did few small mods of existing themes. The problem for me is that I don't have knowledge of QT and most qss modifications I made with trials and errors (I can't find any specification, I just try to edit them by similarity to css, but not all elements work as I expect).
But it's a really good challenge and I think wee should start creating more complex themes (current are very "raw").

**Reply by tsujan** [reply]
I'd make more themes if my free time wasn't taken by coding. Yes, new themes are very welcome.

@nbvcxz 
If you want to make themes, you could do it more easily inside `~/.local/share/lxqt/themes`. Restart the panel/runner from LXQt Session Settings to see how it really looks after each change.

**Comment by Chiitoo** [comment]
Was 'obconf-qt' left out on purpose?

**Reply by tsujan** [reply]
> Was 'obconf-qt' left out on purpose?

Yes. We also removed Openbox's configuration from lxqt-session. However, the last version of `obconf-qt` should be usable and, of course, Openbox can be used with LXQt but with its own config file.

**Reply by Chiitoo** [reply]
Yeah, I noticed, but wasn't sure about the release.  Will need to adjust some things accordingly.

Thanks!
